sda-commons-shared-asyncapi/README.md

```diff
@@ -0,0 +1,131 @@
+One way to use this module is:
+
+* A template file defining the channels using the AsyncAPI spec is part of the API.
+* Definitions for Models classes are generated from code and annotations.
+* This module is used to combine the models and template to a self-contained spec file.
+* The generated AsyncAPI spec is committed into source control.
+
+A manual written AsyncAPI spec template might look like this and can be stored as a resource:
+
+```yaml
+asyncapi: '2.0.0'
+id: 'urn:org:sdase:example:cars'
+defaultContentType: application/json
+
+info:
+  title: Cars Example
+  description: This example demonstrates how to define events around *cars*.
+  version: '1.0.0'
+
+channels:
+  'car-events':
+    publish:
+      operationId: publishCarEvents
+      summary: Car related events
+      description: These are all events that are related to a car
+      message:
+        oneOf:
+          - $ref: '#/components/messages/CarManufactured'
+          - $ref: '#/components/messages/CarScrapped'
+
+components:
+  messages:
+    CarManufactured:
+      title: Car Manufactured
+      description: An event that represents when a new car is manufactured
+      payload:
+        $ref: './schema.json#/definitions/CarManufactured'
+    CarScrapped:
+      title: Car Scrapped
+      description: An event that represents when a car is scrapped
+      payload:
+        $ref: './schema.json#/definitions/CarScrapped'
+```
+
+To automatically generate the AsyncAPI spec and ensure that it is committed to version control, 
+one can use a test like this: 
+
+```java
+    @Test
+    public void generateAndVerifySpec() throws IOException {
+        String expected = AsyncApiGenerator
+            .builder()
+            .withAsyncApiBase(BaseEvent.class.getResource("/asyncapi.yaml"))
+            .withSchema("./schema.json", BaseEvent.class)
+            .generateYaml();
+
+        Path asyncApiPath = Paths.get("./asyncapi.yaml");
+
+        try (AutoCloseableSoftAssertions softly = new AutoCloseableSoftAssertions()) {
+            String actual = new String(Files.readAllBytes(asyncApiPath));
+            Map<String, Object> actualJson =
+                YamlUtil.load(actual, new TypeReference<Map<String, Object>>() {});
+            Map<String, Object> expectedJson =
+                YamlUtil.load(expected, new TypeReference<Map<String, Object>>() {});
+
+            softly.assertThat(actualJson)
+                .as("The current asyncapi.yaml file is not up-to-date. If this happens "
+                    + "locally, just run the test again. The asyncapi.yaml file is updated "
+                    + "automatically after running this test. If this happens in the CI, make sure "
+                    + "that you have committed the latest asyncapi.yaml file!")
+                .isEqualToComparingFieldByFieldRecursively(expectedJson);
+        } finally {
+            Files.write(asyncApiPath, expected.getBytes());
```

Comment:
You should add the encoding to "getBytes".

Reply:
Done